virtual_rainforest/models/animals/animal_communities.py

```diff
@@ -223,20 +249,25 @@ def collect_prey(self, consumer_cohort: AnimalCohort) -> list[AnimalCohort]:
 
             # Filter the potential prey cohorts based on their size
             for cohort in potential_prey_cohorts:
-                if min_size <= cohort.mass <= max_size:
+                if (
+                    min_size <= cohort.mass_current <= max_size
+                    and cohort.individuals != 0
+                    and cohort != consumer_cohort
```

Comment:
I think you want to compare the identity of these objects:

```suggestion
                    and cohort is not consumer_cohort
```